src/django_components/html.py

```diff
@@ -11,91 +10,12 @@ def parse_node(html: str) -> LexborNode:
-    re.IGNORECASE,
-)
-
-HTML_ROOT_TAGS_REVERSE_REGEX = re.compile(
-    r"<doctype_|<html_|</html_|<head_|</head_|<body_|</body_",
-    re.IGNORECASE,
-)
-
-
-html_tag_escapes: Dict[str, str] = {
-    "<!doctype": "<doctype_",
-    "<html": "<html_",
-    "</html": "</html_",
-    "<head": "<head_",
-    "</head": "</head_",
-    "<body": "<body_",
-    "</body": "</body_",
-}
-html_tag_escapes_reverse = {val: key for key, val in html_tag_escapes.items()}
-
-
-class TransformHtmlCallback(Protocol):
-    def __call__(self, head: Optional[LexborNode], body: Optional[LexborNode]) -> None: ...  # noqa: #704
-
-
-# TODO: Remove `transform_html_document`, `insert_before_end`, `html_insert_before_end`
-#       Once Selectolax supports `LaxborNode.insert_child()`
-#       See https://github.com/rushter/selectolax/issues/128
-def transform_html_document(html: str, transform: TransformHtmlCallback) -> str:
-    # Escape <!doctype>, <html>, <head>, and <body> tags, because Selectolax treats them
-    # specially, which makes it impossible to edit them.
-    def on_replace_match(match: "re.Match[str]") -> str:
-        match_str = match[0].lower()
-        return html_tag_escapes[match_str]
-
-    escaped_html = HTML_ROOT_TAGS_REGEX.sub(on_replace_match, html)
-
-    # Selectolax now treats the escaped tags as custom tags, and so Selectolax wraps
-    # the whole content in extra <html><head></head><body> CONTENT </body></html>.
-    # So the actual HTML is under `.body.child`
-    wrapper_tree = LexborHTMLParser(escaped_html).body
-    escaped_tree = wrapper_tree.child if wrapper_tree else None
-    if escaped_tree:
-        body_matches = escaped_tree.css("body_")
-        head_matches = escaped_tree.css("head_")
-        body = body_matches[0] if body_matches else None
-        head = head_matches[0] if head_matches else None
-    else:
-        body = None
-        head = None
-
-    # Finally, we can pass the actual <head> and <body> tags to be transformed
-    # The transformations happen in-place in Selectolax.
-    transform(head, body)
-    transformed_html = cast(str, escaped_tree.html) if escaped_tree else ""
-
-    # After the transformations are applied, we need to un-escape the <doctype_>, <head_>, etc tags
-    def on_reverse_replace_match(match: "re.Match[str]") -> str:
-        match_str = match[0].lower()
-        return html_tag_escapes_reverse[match_str]
-
-    final_html = HTML_ROOT_TAGS_REVERSE_REGEX.sub(on_reverse_replace_match, transformed_html)
-
-    # NOTE: Because of how we work around selectolax to enable modifying the <body> and <head>
-    # tags, selectolax also adds `</doctype_>` at the end of the content when we serialize the HTML.
-    # So we have to remove that.
-    return final_html[: -len("</doctype_>")]
+    # Get all contents of the root
+    root_elem = parser.css_first("root")
+    elems = [*root_elem.iter()] if root_elem else []
+    return cast(LexborNode, root_elem), elems
```

Comment:
Why are we returning the `<root>`-wrapped LexborHTMLParser here? Seems it's not used anywhere, so maybe remove? More elegant to keep the root elem an implementation detail.

Reply:
In this MR there's code only to parse the multi-root HTML, but later on there will be also a need to serialize it back to string. Back when I was young and foolish (few days back), I used the `root` for serializing it. But I can also use just the elements.

Updated in https://github.com/EmilStenstrom/django-components/pull/702/commits/4bff9e73026d7f826dbbde60ef2a3ba85bca36ac!